main/admin.py

```diff
@@ -60,3 +65,21 @@ def get_readonly_fields(self, request, obj=None):
     def get_exclude(self, request, obj=None):
         exclude = tuple(super().get_exclude(request, obj=obj) or ())
         return self._join_and_dedupe(exclude, ("created_on", "updated_on"))
+
+
+# Adapted from solution posted on StackOverflow: https://stackoverflow.com/a/52627264
+class PrettyJSONWidget(widgets.Textarea):
+    """Admin widget class to pretty-print JSONField contents"""
+
+    def format_value(self, value):
+        try:
+            value = json.dumps(json.loads(value), indent=2, sort_keys=True)
+            # these lines will try to adjust size of TextArea to fit to content
+            row_lengths = [len(r) for r in value.split("\n")]
+            self.attrs["rows"] = min(max(len(row_lengths) + 2, 10), 30)
+            self.attrs["cols"] = min(max(max(row_lengths) + 2, 40), 120)
+            self.attrs["style"] = "font-family: monospace"
+            return value
+        except Exception as e:  # pylint: disable=broad-except
```

Comment:
What kind of exceptions are we expecting here? I'm also wondering if we should `log.exception` below instead of `log.warning`

Reply:
I want to avoid sending errors to Sentry if something just goes wrong in these admin forms. As far as exceptions go, I guess the obvious one is if someone sets the underlying value in the db to a non-JSON value. Personally I think it's OK just to default to a normal textarea if something in the `try` block fails for any reason

Reply:
That's fine with me